turbopack/crates/turbo-tasks/src/manager.rs

```diff
@@ -1450,6 +1491,16 @@ impl<B: Backend + 'static> TurboTasksBackendApi<B> for TurboTasks<B> {
     unsafe fn reuse_transient_task_id(&self, id: Unused<TaskId>) {
         unsafe { self.transient_task_id_factory.reuse(id.into()) }
     }
+
+    fn read_task_state_boxed(&self, func: Box<dyn FnOnce(&B::TaskState) + '_>) {
```

Comment:
I feel like this Box can be avoided via `func: &mut dyn FnMut(&B::TaskState) + '_`

Reply:
I agree this box sucks, as this is likely a hot codepath. Requiring `FnMut` might lead to more cloning/`Arc`s/etc at the callsites, which is probably worse.

I'll see what I can do.

Reply:
Good news! I got rid of it. I was able to use `&mut dyn FnMut(...)` along with `Option::take` to get ownership of `func`, so that the `Ext` trait version can still accept `FnOnce`.